<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NoiseFigureGUI</class>
 <widget class="RollupContents" name="NoiseFigureGUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>404</width>
    <height>693</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>352</width>
    <height>0</height>
   </size>
  </property>
  <property name="font">
   <font>
    <family>Liberation Sans</family>
    <pointsize>9</pointsize>
   </font>
  </property>
  <property name="focusPolicy">
   <enum>Qt::StrongFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Noise Figure</string>
  </property>
  <widget class="QWidget" name="settingsContainer" native="true">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>390</width>
     <height>181</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>350</width>
     <height>0</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Settings</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>3</number>
    </property>
    <property name="leftMargin">
     <number>2</number>
    </property>
    <property name="topMargin">
     <number>2</number>
    </property>
    <property name="rightMargin">
     <number>2</number>
    </property>
    <property name="bottomMargin">
     <number>2</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="powLayout">
      <property name="topMargin">
       <number>2</number>
      </property>
      <item>
       <widget class="QLabel" name="deltaFrequencyLabel">
        <property name="minimumSize">
         <size>
          <width>16</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Df</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="ValueDialZ" name="deltaFrequency" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>32</width>
          <height>16</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Liberation Mono</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="focusPolicy">
         <enum>Qt::StrongFocus</enum>
        </property>
        <property name="toolTip">
         <string>Measurement frequency from center in Hz</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="deltaUnits">
        <property name="text">
         <string>Hz </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="channelPowerLayout">
        <item>
         <widget class="QLabel" name="channelPower">
          <property name="toolTip">
           <string>Channel power</string>
          </property>
          <property name="layoutDirection">
           <enum>Qt::RightToLeft</enum>
          </property>
          <property name="text">
           <string>0.0</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="channelPowerUnits">
          <property name="text">
           <string> dB</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="powerLayout">
      <item>
       <widget class="QLabel" name="channelPowerMeterUnits">
        <property name="text">
         <string>dB</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="LevelMeterSignalDB" name="channelPowerMeter" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Liberation Mono</family>
          <pointsize>8</pointsize>
         </font>
        </property>
        <property name="toolTip">
         <string>Level meter (dB) top trace: average, bottom trace: instantaneous peak, tip: peak hold</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line_5">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="phySettingsLayout">
      <item>
       <widget class="QLabel" name="fftSizeLabel">
        <property name="text">
         <string>FFT Size</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="fftSize">
        <property name="minimumSize">
         <size>
          <width>50</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>FFT size (number of bins)</string>
        </property>
        <property name="currentText">
         <string>64</string>
        </property>
        <item>
         <property name="text">
          <string>64</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>128</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>256</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>512</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>1k</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line_4">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="rfBWLabel">
        <property name="text">
         <string>BW</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="rfBWText">
        <property name="minimumSize">
         <size>
          <width>30</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Measurement bandwidth as determined by width of a single FFT bin</string>
        </property>
        <property name="text">
         <string>10.0k</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="fftCountLabel">
        <property name="text">
         <string>FFTs to average</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="fftCount">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Number of FFT outputs to average to calculate power</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>10</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="pageStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="fftCountText">
        <property name="minimumSize">
         <size>
          <width>30</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>2.4k</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line_8">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line_7">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line_6">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="parameterLabel">
        <property name="text">
         <string>Sweep</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="setting">
        <property name="minimumSize">
         <size>
          <width>160</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Device parameter to sweep</string>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
        <item>
         <property name="text">
          <string>centerFrequency</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>gain</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="freqLayout">
      <item>
       <widget class="QComboBox" name="frequencySpec">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>68</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>How to specify frequencies to measure NF at</string>
        </property>
        <item>
         <property name="text">
          <string>Range</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Step</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>List</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="startLabel">
        <property name="text">
         <string>Start</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="start">
        <property name="toolTip">
         <string>Start frequency in MHz</string>
        </property>
        <property name="decimals">
         <number>3</number>
        </property>
        <property name="maximum">
         <double>10000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="stopLabel">
        <property name="text">
         <string>Stop</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="stop">
        <property name="toolTip">
         <string>Stop frequency in MHz</string>
        </property>
        <property name="decimals">
         <number>3</number>
        </property>
        <property name="maximum">
         <double>10000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="stepsLabel">
        <property name="text">
         <string>Steps</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="steps">
        <property name="toolTip">
         <string>Number of frequencies to measure at between start and stop inclusive</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="stepLabel">
        <property name="text">
         <string>Step</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="step">
        <property name="toolTip">
         <string>Frequency step in MHz</string>
        </property>
        <property name="decimals">
         <number>3</number>
        </property>
        <property name="minimum">
         <double>0.001000000000000</double>
        </property>
        <property name="maximum">
         <double>10000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="list">
        <property name="toolTip">
         <string>List of frequencies in MHz to measure NF at</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="filterLayout">
      <item>
       <widget class="ButtonSwitch" name="startStop">
        <property name="toolTip">
         <string>Start/stop noise figure measurement</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../../sdrgui/resources/res.qrc">
          <normaloff>:/play.png</normaloff>
          <normalon>:/stop.png</normalon>:/play.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="saveResults">
        <property name="toolTip">
         <string>Save results to a .csv file</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../../sdrgui/resources/res.qrc">
          <normaloff>:/save.png</normaloff>:/save.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="clearResults">
        <property name="toolTip">
         <string>Clear results</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../../sdrgui/resources/res.qrc">
          <normaloff>:/bin.png</normaloff>:/bin.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QToolButton" name="enr">
        <property name="toolTip">
         <string>Open ENR dialog</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../../sdrgui/resources/res.qrc">
          <normaloff>:/listing.png</normaloff>:/listing.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="control">
        <property name="toolTip">
         <string>Open noise source control dialog</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../../sdrgui/resources/res.qrc">
          <normaloff>:/link.png</normaloff>:/link.png</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="resultsContainer" native="true">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>500</y>
     <width>391</width>
     <height>171</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="windowTitle">
    <string>Results</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayoutTable">
    <property name="spacing">
     <number>2</number>
    </property>
    <property name="leftMargin">
     <number>3</number>
    </property>
    <property name="topMargin">
     <number>3</number>
    </property>
    <property name="rightMargin">
     <number>3</number>
    </property>
    <property name="bottomMargin">
     <number>3</number>
    </property>
    <item>
     <widget class="QTableWidget" name="results">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Received packets</string>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
      <column>
       <property name="text">
        <string>Freq (MHz)</string>
       </property>
       <property name="toolTip">
        <string>Frequency in MHz the noise figure was measured at</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>NF (dB)</string>
       </property>
       <property name="toolTip">
        <string>Noise figure in dB</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>T (K)</string>
       </property>
       <property name="toolTip">
        <string>Noise temperature in Kelvin</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Y (dB)</string>
       </property>
       <property name="toolTip">
        <string>Y factor in dB</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>ENR (dB)</string>
       </property>
       <property name="toolTip">
        <string>Excess noise ratio of the noise source in dB</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Floor (dBm)</string>
       </property>
       <property name="toolTip">
        <string>Noise floor in dBm for 1Hz bandwidth at 290K</string>
       </property>
      </column>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="chartContainer" native="true">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>210</y>
     <width>391</width>
     <height>268</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>0</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Chart</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0">
    <property name="spacing">
     <number>2</number>
    </property>
    <property name="leftMargin">
     <number>3</number>
    </property>
    <property name="topMargin">
     <number>3</number>
    </property>
    <property name="rightMargin">
     <number>3</number>
    </property>
    <property name="bottomMargin">
     <number>3</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QComboBox" name="chartSelect">
        <property name="toolTip">
         <string>Select data to plot on chart</string>
        </property>
        <item>
         <property name="text">
          <string>Noise Figure (dB)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Noise Temperature (K)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Y (dB)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>ENR (dB)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Noise floor (dBm)</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="openReference">
        <property name="toolTip">
         <string>Open .csv file with reference data</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../../sdrgui/resources/res.qrc">
          <normaloff>:/load.png</normaloff>:/load.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="clearReference">
        <property name="toolTip">
         <string>Clear reference data</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../../sdrgui/resources/res.qrc">
          <normaloff>:/bin.png</normaloff>:/bin.png</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QChartView" name="chart">
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>250</height>
       </size>
      </property>
      <property name="toolTip">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ButtonSwitch</class>
   <extends>QToolButton</extends>
   <header>gui/buttonswitch.h</header>
  </customwidget>
  <customwidget>
   <class>RollupContents</class>
   <extends>QWidget</extends>
   <header>gui/rollupcontents.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header>QtCharts</header>
  </customwidget>
  <customwidget>
   <class>ValueDialZ</class>
   <extends>QWidget</extends>
   <header>gui/valuedialz.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>LevelMeterSignalDB</class>
   <extends>QWidget</extends>
   <header>gui/levelmeter.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>deltaFrequency</tabstop>
  <tabstop>fftSize</tabstop>
  <tabstop>fftCount</tabstop>
  <tabstop>frequencySpec</tabstop>
  <tabstop>start</tabstop>
  <tabstop>stop</tabstop>
  <tabstop>steps</tabstop>
  <tabstop>step</tabstop>
  <tabstop>list</tabstop>
  <tabstop>startStop</tabstop>
  <tabstop>saveResults</tabstop>
  <tabstop>clearResults</tabstop>
  <tabstop>enr</tabstop>
  <tabstop>control</tabstop>
  <tabstop>chartSelect</tabstop>
  <tabstop>openReference</tabstop>
  <tabstop>clearReference</tabstop>
  <tabstop>chart</tabstop>
 </tabstops>
 <resources>
  <include location="../../../sdrgui/resources/res.qrc"/>
 </resources>
 <connections/>
</ui>
